Ecal/src/Ecal/EcalRawDecoder.cxx

```diff
@@ -195,29 +198,32 @@ void EcalRawDecoder::produce(framework::Event& event) {
            * 10101010 | BXID (12) | WADD (9) | 1010
            */
           // std::cout << " : ROC Header";
+          /*
+           * These are unused, should they be? FIXME
           link_crc << w;
           uint32_t bx_id = (w >> 16) & packing::utility::mask<12>;
           uint32_t short_event = (w >> 10) & packing::utility::mask<6>;
           uint32_t short_orbit = (w >> 7) & packing::utility::mask<3>;
           uint32_t hamming_errs = (w >> 4) & packing::utility::mask<3>;
+          */
         } else if (channel_id == common_mode_channel) {
           /** Common Mode Channels
            * 10 | 0000000000 | Common Mode ADC 0 (10) | Common Mode ADC 1 (10)
            */
           link_crc << w;
           // std::cout << " : Common Mode";
         } else if (channel_id == 39) {
-          // CRC checksum from ROC
-          uint32_t crc = w;
-          // std::cout << " : CRC checksum  : " << hex(link_crc.get()) <<
-          // " =? " << hex(crc);
           /*
-          if (link_crc.get() != crc) {
-            EXCEPTION_RAISE("BadCRC",
-                            "Our calculated link checksum doesn't match the "
-                            "one from raw data.");
-          }
-          */
+    // CRC checksum from ROC
+    uint32_t crc = w;
+    // std::cout << " : CRC checksum  : " << hex(link_crc.get()) <<
+    // " =? " << hex(crc);
+    if (link_crc.get() != crc) {
+      EXCEPTION_RAISE("BadCRC",
+                      "Our calculated link checksum doesn't match the "
+                      "one from raw data.");
+    }
+    */
```

Comment:
No, I can't do a suggestion since all the lines are deleted, but keeping this there in comment is helpful when the HGCROC is sending along its own CRC checksum of the data for us to compare against.

We had to comment it out because the v2 HGCROC we were using was not sending a checksum and thus the check would always fail.